src/mono/wasm/features.md

```diff
@@ -2,36 +2,311 @@
+
+## Browser application template
+
+You can create simple application template by running `dotnet new wasmbrowser`.
+
+Then you could `dotnet run` and open the URL which it printed to test the app in your browser. For example `http://localhost:5292/index.html`
+
+You can also `dotnet publish -c Release` which will publish your app to [AppBundle](#Project-folder-structure) folder.
+
+## JavaScript interoperability
+
+When you want to call JavaScript functions from C# or managed code from JavaScript, you can annotate static C# methods with `[JSImport]` or `[JSExport]` attributes.
+
+You can read [blog article](https://devblogs.microsoft.com/dotnet/use-net-7-from-any-javascript-app-in-net-7/) about it.
+
+You can also explore simple [TODO app sample](https://github.com/pavelsavara/dotnet-wasm-todo-mvc).
+
+For more details please see [documentation](https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/import-export-interop).
+
+## Embedding dotnet in existing JavaScript applications
+
+The npm tool chains usual in the JavaScript community are not well integrated with the MSBuild tool chain of the dotnet.
+Your project will need to handle both build systems.
+
+You will need pass output of the MSBuild from the `AppBundle` folder for the npm toolchain to process and host it on your web server with the other assets of your application.
+
+You can also explore simple sample of dotnet as [React component](https://github.com/maraf/dotnet-wasm-react).
+
+# Downloaded assets
+
+## Project folder structure
+
+Described as relative to simple application project.
+
+- `./wwwroot` is optional project folder, into which you can place files which should be also deployed to the web server.
+- `./bin/Release/net8.0/browser-wasm/AppBundle` folder which should be hosted by the HTTP server.
+- `./bin/Release/net8.0/browser-wasm/AppBundle/index.html` - the page which is hosting the application.
+- `./bin/Release/net8.0/browser-wasm/AppBundle/main.js` - typically the main JavaScript entry point, it will `import { dotnet } from './_framework/dotnet.js'` and `await dotnet.run();`.
+- `./bin/Release/net8.0/browser-wasm/AppBundle/_framework` - contains all the assets of the runtime and the managed application.
+
+You can flatten the `_framework` folder away by `<WasmRuntimeAssetsLocation>./</WasmRuntimeAssetsLocation>` in the project file.
+
+## `_framework` folder structure
+- `dotnet.js` - is the main entrypoint with the [JavaScript API](#JavaScript-API). It will load the rest of the runtime.
+- `dotnet.native.js` - is posix emulation layer by [emscripten](https://github.com/emscripten-core/emscripten) project
+- `dotnet.runtime.js` - is integration of the dotnet with the browser
+- `blazor.boot.json` - contains list of all other assets and their integrity hash and also various configuration flags.
```

Comment:
```suggestion
- `blazor.boot.json` - contains list of all other assets and their integrity hash and also various configuration flags. Note: even though this file has `blazor` in its name, it is completely independent of `Blazor`.
```

Reply:
At this point I'm not sure I love that we have that file name. I would not elaborate it in the doc.